### PR TITLE
fix: [Inheritance] Invalid code generation when base class have parameters with default values

### PR DESCRIPTION
**Pull request by marcelomendoncasoares**
This PR closes the issue #3082, where the problem is described in detail. As a brief summary, when using the model inheritance experimental feature, there is a bug if any of the base class fields have default values. The generated constructor tries to initialize the default fields as if they belong to the class itself instead of delegating to super.

## Pre-launch Checklist

- [x] I read the [Contribute](https://docs.serverpod.dev/contribute) page and followed the process outlined there for submitting PRs.
- [x] This update contains only one single feature or bug fix and nothing else. (If you are submitting multiple fixes, please make multiple PRs.)
- [x] I read and followed the [Dart Style Guide](https://dart.dev/guides/language/effective-dart/style) and formatted the code with [dart format](https://dart.dev/tools/dart-format).
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`), and made sure that the documentation follows the same style as other Serverpod documentation. I checked spelling and grammar.
- [x] I added new tests to check the change I am making.
- [x] All existing and new tests are passing.
- [x] Any breaking changes are documented below.

If you need help, consider asking for advice on the [discussion board](https://github.com/serverpod/serverpod/discussions).

## Breaking changes

No breaking changes, only the fix for a case not covered on the tests.

### CODE REVIEW

File: tools/serverpod_cli/test/generator/dart/client_code_generator/inheritance_class_test.dart [hunk 1]
```diff
@@ -35,31 +35,35 @@ void main() {
   );
 
   group(
-      'Given a child-class named $childClassName with one primitive var extending a parent-class named $parentClassName with one primitive var when generating code',
+      'Given a child-class named $childClassName with one primitive var and a var with default value extending a parent-class named $parentClassName with one primitive var and a var with default value when generating code',
```
Comment (SandPod) [inline tools/serverpod_cli/test/generator/dart/client_code_generator/inheritance_class_test.dart:38]: Suggestion: Break this test out into a separate test and add a similar test for the `server_code_generator`.

We try to isolate each test so that there is a single reason for failure. We believe this will guide future developers when implementing new features. Please break the test and setup validating the changes in this PR out into a separate group to make the setup as simple as possbile to validate the test scenario.

We also want to validate that this is working correctly for both the server and client generator. Regardless of how we refactor the code.

**Review by SandPod — CHANGES_REQUESTED** [review]
Awesome bug report and nice and isolated fix.

Apart from the comment on the tests, could you please add a model in the test project that validates this scenario? We generally try to have each feature of our models generated once in the project to catch edge cases.

File: tools/serverpod_cli/test/generator/dart/client_code_generator/inheritance_class_test.dart (hunk 1 above)
Comment (marcelomendoncasoares) [reply tools/serverpod_cli/test/generator/dart/client_code_generator/inheritance_class_test.dart]: Done! Following the hint from the other PR, I also simplified the test to remove validations that would become redundant when separating the tests.

**Comment by marcelomendoncasoares** [comment]
> Awesome bug report and nice and isolated fix.
> 
> Apart from the comment on the tests, could you please add a model in the test project that validates this scenario? We generally try to have each feature of our models generated once in the project to catch edge cases.

No problem. Just added two models (the parent and the child) with the exact same configuration from the server and client tests.
